packages/server/src/integrations/elasticsearch.ts

```diff
@@ -81,7 +99,23 @@ class ElasticSearchIntegration implements IntegrationBase {
 
   constructor(config: ElasticsearchConfig) {
     this.config = config
-    this.client = new Client({ node: config.url })
+
+    let newConfig = {
```

Comment:
Hm! I initially attempted this approach which I do agree looks cleaner, but I came up against type errors.

This is the output I get based on edits against my current branch. Perhaps I missed some sort of declaration in another part of the code would allow me to avoid the type errors?

![1](https://user-images.githubusercontent.com/79575739/198581016-107df2b1-eee7-4a94-ba21-87956f461b90.jpg)

![2](https://user-images.githubusercontent.com/79575739/198581037-bab8e297-a08c-44fe-bc63-6c322b389d8c.jpg)

Reply:
In this case we can update the import from the elasticsearch dependency from:
```ts
const { Client } = require("@elastic/elasticsearch")
```
To:
```ts
import { Client, ClientOptions } from "@elastic/elasticsearch"
```

This now gives us access to the types provided by the elasticsearch dep. 

Now we can use the `ClientOptions` type by assigning the type to `clientConfig`

```ts
    const clientConfig: ClientOptions = {
      node: this.config.url
    }
```

This should resolve the type error and also make our configuration more type safe.

Reply:
Ahhh, got it! I've cleaned up the constructor and re-tested the various possible configurations.